### PR TITLE
fix daily profit data bug and daily profit curve example

### PR DESCRIPTION
**Pull request by octaviusgus**
## Summary
BUG:
Backtest includes data from 2018-08-12 to 2021-07-05
Yet daily profits, in this case, returned 2018-09-23 to 2021-05-19
Also different list lengths
Bug is present because I thought daily_profit included each backtested day but when I tried plotting equity curve for another strategy with less buy signals I realized daily_profit could exclude days from the beggining and the end of the backtest period. This was a problem because since daily_profit returned just a list of profit values (smaller than the backtest period), there is no way to know what index corresponds to what date.
Fixed by adding date to the daily_profit data returned from optimize_reports.generate_daily_stats

```
2021-07-06 22:19:51,617 - freqtrade.optimize.backtesting - INFO - Backtesting with data from 2018-08-12 08:00:00 up to 2021-07-05 00:00:00 (1057 days).

PRINT DAILY_PROFIT LIST FROM optimize_reports.generate_daily_stats:

close_date
2018-09-23 00:00:00+00:00   -0.003287
2018-09-24 00:00:00+00:00    0.000000
2018-09-25 00:00:00+00:00    0.312506
2018-09-26 00:00:00+00:00    0.000000
2018-09-27 00:00:00+00:00    0.000000
                               ...   
2021-05-15 00:00:00+00:00    0.000000
2021-05-16 00:00:00+00:00   -0.812115
2021-05-17 00:00:00+00:00    0.000000
2021-05-18 00:00:00+00:00    0.000000
2021-05-19 00:00:00+00:00   -0.002082
Freq: D, Name: profit_abs, Length: **970**, dtype: float64
```

## Quick changelog

- Export daily profits as tuple (date, profit)
- Adapt the jupyter notebook snippet

## What's new?

Equity curve in the notebook works

### CODE REVIEW

File: freqtrade/templates/strategy_analysis_example.ipynb [hunk 1]
```diff
@@ -215,13 +215,18 @@
     "stats = load_backtest_stats(backtest_dir)\n",
     "strategy_stats = stats['strategy'][strategy]\n",
     "\n",
+    "dates = []\n",
+    "profits = []\n",
+    "for date_profit in strategy_stats['daily_profit']:\n",
```
Comment (xmatthias) [inline freqtrade/templates/strategy_analysis_example.ipynb:220]: When doing updates to the notebook, please always run the 2 jupyter commands afterwards (as documented [here](https://www.freqtrade.io/en/stable/developer/#updating-example-notebooks)).

that'll keep the notebook "clean" (it's clean now too - but this makes sure of it) - and also makes it available as documentation should someone want to copy it into an existing notebook.

Comment (octaviusgus) [reply freqtrade/templates/strategy_analysis_example.ipynb:220]: I ran the commands but I'm not seeing any changes in any files, am I doing something wrong?

Comment (octaviusgus) [reply freqtrade/templates/strategy_analysis_example.ipynb:220]: Should I do anything else for the PR?

Comment (octaviusgus) [reply freqtrade/templates/strategy_analysis_example.ipynb:220]: Oh I see you ran the commands to get the docs updated, sorry I couldn't look at this earlier.

**Review by xmatthias — APPROVED** [review]
LGTM :+1:

**Pull request merged**